src/Microsoft.OpenApi.Hidi/OpenApiSpecVersionHelper.cs

```diff
@@ -14,17 +13,30 @@ public static OpenApiSpecVersion TryParseOpenApiSpecVersion(string value)
             {
                 throw new InvalidOperationException("Please provide a version");
             }
-            var res = value.Split('.', StringSplitOptions.RemoveEmptyEntries).FirstOrDefault();
+            // Split the version string by the dot
+            var versionSegments = value.Split('.', StringSplitOptions.RemoveEmptyEntries);
 
-            if (int.TryParse(res, out var result))
+            if (!int.TryParse(versionSegments[0], out var majorVersion)
+                || !int.TryParse(versionSegments[1], out var minorVersion))
             {
-                if (result is >= 2 and < 3)
-                {
-                    return OpenApiSpecVersion.OpenApi2_0;
-                }
+                throw new InvalidOperationException("Invalid version format. Please provide a valid OpenAPI version (e.g., 2.0, 3.0, 3.1).");
```

Comment:
Can we also add another check like the previous one to validate that the major version is between 2 and 3?